src/TinySegmenter.jl

```diff
@@ -83,45 +82,66 @@ const Hchar = UInt8('H')
 const Ochar = UInt8('O')
 const Uchar = UInt8('U')
 const Bchar = UInt8('B')
-function _ctype(c::Char)
+function ctype(c::Char)
   return get(CHARDICT, c, '一' <= c <= '龠' ? Hchar : Ochar)
 end
 
-function tokenize(text::AbstractString)
-  if isempty(text)
-    return US[]
-  end
+"""
+    tokenize(text::AbstractString)
 
-  result = US[]
-  segment = [B3, B2, B1]
-  ctype = UInt8['O', 'O', 'O']
-  for char in text
-    push!(segment, char)
-    push!(ctype, _ctype(char))
-  end
+Given a `text` string, `tokenize` attempts to segment it into a list
+of words, and in particular tries to segment Japanese Kanji text
+into words ("tokens" or "segments"), using the heuristic TinySegmenter algorithm.
```

Comment:
`heuristic` is not suitable. TinySegmenter is model based segmenter. It includes ada boosting based model within this code.